src/test/java/ast/ASTNodesKtTest.kt

```diff
@@ -0,0 +1,173 @@
+package ast
+
+import antlr.TreepatVisitorImplementation
+import antlr.generate.TreepatLexer
+import antlr.generate.TreepatParser
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+import org.antlr.v4.runtime.tree.ParseTree
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+internal class ASTNodesKtTest {
+
+    private var countNode: Int = 0
+    private var countChild: Int = 0
+    private var countSibling: Int = 0
+
+    @BeforeEach
+    fun setUp() {
+        countNode = 0
+        countChild = 0
+        countSibling = 0
+    }
+
+    private fun createParserTree(inputAntlrString: String): ParseTree {
+        val lexer = TreepatLexer(CharStreams.fromString(inputAntlrString))
+        val tokenStream = CommonTokenStream(lexer)
+        val parser = TreepatParser(tokenStream)
+        return parser.subtree()
+    }
+
+    private fun countNodes(node: ASTNode) {
+        when (node) {
+            is Node -> countNode++
+            is Child -> {
+                countNodes(node.father)
+                countNodes(node.child)
+                countChild++
+            }
+            is Sibling -> {
+                node.siblings.map { countNodes(it) }
+                countSibling++
+            }
+            else -> throw IllegalArgumentException("This ASTNode subtype is not supported.")
+        }
+    }
+
+    @Test
+    fun `should have one node and then up`() {
+        // arrange input string
+        val inputAntlrString = """
+            A
+        """.trimIndent()
+        // create parser tree
+        val tree = createParserTree(inputAntlrString)
+        // create visitor
+        val visitor = TreepatVisitorImplementation()
+        val root = visitor.visit(tree)
+        // get count of nodes
+        countNodes(root)
+        // assert
+        assertEquals(countNode, 1)
+        assertEquals(countChild, 0)
+        assertEquals(countSibling, 0)
```

Comment:
I think is a good idea to add comments to the stages of the test but lets just add the stages arrange, act and assert. Like in the file `ChildFunctionKtTest.kt`.
```suggestion
        // arrange
        val inputAntlrString = """
            A
        """.trimIndent()
        val tree = createParserTree(inputAntlrString)
        val visitor = TreepatVisitorImplementation()
        val root = visitor.visit(tree)
        // act
        countNodes(root)
        // assert
        assertEquals(countNode, 1)
        assertEquals(countChild, 0)
        assertEquals(countSibling, 0)
```
But is up to you. 😃